rlclientlib/utility/apim_http_authorization.cc

```diff
@@ -1,17 +1,33 @@
 #include "apim_http_authorization.h"
+#include <boost/algorithm/string.hpp>
 
 namespace reinforcement_learning {
   int apim_http_authorization::init(const utility::configuration& config, api_status* status, i_trace* trace) {
     const auto api_key = config.get(name::HTTP_API_KEY, nullptr);
     if (api_key == nullptr) {
       RETURN_ERROR(trace, status, http_api_key_not_provided);
     }
+    const auto auth_type = config.get(name::HTTP_KEY_TYPE, nullptr);
+    if (auth_type == nullptr) {
+        RETURN_ERROR(trace, status, http_auth_type_not_provided);
+    }
+
     _api_key = api_key;
+    _auth_type = auth_type;
+
     return error_code::success;
   }
 
   int apim_http_authorization::get_http_headers(http_headers& headers, api_status* status) {
-    headers.add(_XPLATSTR("Ocp-Apim-Subscription-Key"), _api_key.c_str());
+      if (boost::iequals(_auth_type, "bearer"))
```

Comment:
Is it possible to define an enum authType which has 2 values: token or apiKey?

Reply:
the key type user gives can be case insensitive. Considering that using enum seems not feasible